pkg/builder/TwoSampleTTestBuilder.go

```diff
@@ -0,0 +1,332 @@
+(for a 95% minor threshold) the result is a 1. A P-value greater than the minor threshold
+will cause a return of 0.
+*/
+import (
+	"errors"
+	"fmt"
+	"github.com/aclements/go-moremath/stats"
+	"github.com/go-playground/validator/v10"
+	"log"
+	"strings"
+)
+
+// use a single instance of Validate, it caches struct info
+
+// setters:
+// The goodnessPolarity indicates if this population is positive good (like for TS/CSI)
+// or negative good like for RMSE or BIAS. The null hypothesis is that the populations
+// are identical so a positive difference for an error value is bad
+// and a negative difference is good (less error in the experimental population
+// is good for error). If the statistic is Threat Score / Critical
+// Succes Index then a positive difference is good (0 is the worst 1 is the best)
+// and a negative difference in the experimental population would be bad.
+// This information is determined outside of this builder (the builder doesn't know
+// what parameter combination is being tested) so the builder must be told
+// what the "goodnes polarity" is +1 or -1.
+func (scc *ScorecardCell) SetGoodnessPolarity(polarity GoodnessPolarity) error {
+	errs := validate.Var(polarity, "required,oneof=-1 1")
+	if errs != nil {
+		log.Print(errs)
+		return errors.New(fmt.Sprint("TwoSampleTTestBuilder SetGoodnessPolarity", errs))
+	} else {
+		scc.goodnessPolarity = polarity
+	}
+	return nil // no errors
+}
+
+// set the major p-value threshold
+func (scc *ScorecardCell) SetMajorThreshold(threshold Threshold) error {
+	if errs := validate.Var(threshold, "required,gt=0,lt=.5"); errs != nil {
+		log.Print(errs)
+		return errors.New(fmt.Sprint("TwoSampleTTestBuilder SetMajorThreshold", errs))
+	} else {
+		scc.majorThreshold = threshold
+	}
+	return nil // no errors
+}
+
+// set the major p-value threshold
+func (scc *ScorecardCell) SetMinorThreshold(threshold Threshold) error {
+	if errs := validate.Var(threshold, "required,gt=0,lt=.5"); errs != nil {
+		log.Print(errs)
+		return errors.New(fmt.Sprint("TwoSampleTTestBuilder SetMinorThreshold", errs))
+	} else {
+		scc.minorThreshold = threshold
+	}
+	return nil // no errors
+}
+
+// get the return value based on the major and minor thresholds compared to the p-value
+func deriveValue(scc *ScorecardCell, difference float64, pval float64) (int, error) {
+	if errs := validate.Var(difference, "required"); errs != nil {
+		log.Print(errs)
+		return 0, errors.New(fmt.Sprint("TwoSampleTTestBuilder deriveValue", errs))
+	} else {
+		if errs := validate.Var(pval, "required"); errs != nil {
+			fmt.Println(errs)
+			return -9999, errors.New(fmt.Sprint("TwoSampleTTestBuilder deriveValue", errs))
+		} else {
+			if pval <= float64(scc.majorThreshold) {
+				return 2 * int(scc.goodnessPolarity), nil
+			}
+			if pval <= float64(scc.minorThreshold) {
+				return 1 * int(scc.goodnessPolarity), nil
+			}
+			return 0, nil
+		}
+	}
```

Comment:
I forgot where we left the discussion around Go's Zero values. Should the second return here be `0`? It feels like it and the first return should be the same. The caller would then be responsible for checking `deriveValue`'s returned error and skipping adding the value to the scorecard cell.

Additionally, can we un-nest this function? It's easier to figure out what's going on when flattened out.

```suggestion
	if errs := validate.Var(difference, "required"); errs != nil {
		log.Print(errs)
		return 0, errors.New(fmt.Sprint("TwoSampleTTestBuilder deriveValue", errs))
	}

	if errs := validate.Var(pval, "required"); errs != nil {
		fmt.Println(errs)
		return -9999, errors.New(fmt.Sprint("TwoSampleTTestBuilder deriveValue", errs))
	}

	if pval <= float64(scc.majorThreshold) {
		return 2 * int(scc.goodnessPolarity), nil
	}

	if pval <= float64(scc.minorThreshold) {
		return 1 * int(scc.goodnessPolarity), nil
	}

	return 0, nil
```

Lastly, I'm curious why ScorecardCell is being passed into `deriveValue` instead of this being a private method of a ScorecardCell.